adlik_serving/runtime/ml/algorithm/grid/grid_output.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 ZTE corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ADLIK_SERVING_RUNTIME_ML_ALGORITHM_GRID_GRID_OUTPUT_H
+#define ADLIK_SERVING_RUNTIME_ML_ALGORITHM_GRID_GRID_OUTPUT_H
+
+#include <vector>
+
+#include "adlik_serving/runtime/ml/algorithm/grid/cell.h"
+#include "adlik_serving/runtime/ml/algorithm/grid/types.h"
+
+namespace ml_runtime {
+
+struct GridInput;
+struct Neighbors;
+
+struct GridOutput {
+  GridOutput();
+  GridOutput(const Neighbors& neighbors,
+             const Rsrp serverRSRP_core,
+             const Rsrp neighRSRP_core1,
+             const Rsrp neighRSRP_core2);
+
+  void update(const GridInput&);
+  void arrangeStats();
+
+  template <typename F>
+  void accept(F&& f) const {
+    std::forward<F>(f)(this->serverRSRP_core);
+    std::forward<F>(f)(this->serverRSRP_max);
```

Comment:
`std::forward` one object multiple times could be dangerous, since the content of the object could been moved after its first usage, and the second usage might expect a not-moved value.